skore/src/skore/project/metadata.py

```diff
@@ -37,6 +38,109 @@ class Metadata(DataFrame):
 
     _metadata = ["project"]
 
+    @staticmethod
+    def _explore_sk_pipeline(
+        pipeline: Pipeline, itemized_pipeline: list, depth: int = 0, max_depth: int = 5
+    ) -> list:
+        """
+        Explore the pipeline and return a list of estimators or transformers.
+
+        Parameters
+        ----------
+        pipeline : Pipeline
+            The pipeline to explore.
+
+        Returns
+        -------
+        list
+            A list of estimators or transformers in the pipeline.
+        """
+        if depth > max_depth:
+            return itemized_pipeline
+
+        for _, step in pipeline.steps:
+            type_ = type(step)
+            if type_ == Pipeline:
+                itemized_pipeline = Metadata._explore_sk_pipeline(
+                    step, itemized_pipeline, depth=depth + 1
+                )
+            elif type_ == FeatureUnion:
+                itemized_pipeline = Metadata._explore_sk_feature_union(
+                    step, itemized_pipeline, depth=depth + 1
+                )
+            elif str(type_) == "<class 'skrub._table_vectorizer.TableVectorizer'>":
```

Comment:
I'm not sure it's how it should be done, but I see no other way without importing skrub, which we don't want to have mandatory in our imports.